src/promptflow-devkit/promptflow/batch/_batch_engine.py

```diff
@@ -184,13 +194,19 @@ def run(
                                 "Current thread is not main thread, skip signal handler registration in BatchEngine."
                             )
 
-                    # set batch input source from input mapping
-                    set_batch_input_source_from_inputs_mapping(inputs_mapping)
-                    # if using eager flow, the self._flow is none, so we need to get inputs definition from executor
-                    inputs = self._executor_proxy.get_inputs_definition() if self._is_eager_flow else self._flow.inputs
-                    # resolve input data from input dirs and apply inputs mapping
-                    batch_input_processor = BatchInputsProcessor(self._working_dir, inputs, max_lines_count)
-                    batch_inputs = batch_input_processor.process_batch_inputs(input_dirs, inputs_mapping)
+                    if self._executor_proxy.should_apply_inputs_mapping:
```

Comment:
why do we need a bool indicator like this? 
Is this case, input_mapping is None here?